docs/operator-manual/server-commands/argocd-server.md

```diff
@@ -0,0 +1,66 @@
+## argocd-server
+
+Run the argocd API server
```

Comment:
If we are exposing the server parameters commands in the docs, I think we should tweak the description or synopsis to make it more obvious that this is a server process, something like this:

"This command starts the Argo CD internal API server, which is a gRPC/REST server which exposes the API consumed by the Web UI, CLI, and CI/CD systems. This server may be customized via the command parameters below." 
(Most of this sentence was borrowed from https://argoproj.github.io/argo-cd/operator-manual/architecture/)

(Also: I'm adding the review comment to the markdown, but of course it would need to go into the command itself)

Reply:
Updated.